test/runtests.py

```diff
@@ -7,43 +7,64 @@
 import sys
 import os
 import subprocess as SP
+import argparse, textwrap
+import xml.etree.ElementTree as ET
+
+parser = argparse.ArgumentParser(
+    description='ChakraCore *nix Test Script',
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=textwrap.dedent('''\
+        Samples:
+
+        test all folders:
+            {0}
+
+        test only Array:
+            {0} Array
+
+        test a single file:
+            {0} Basics/hello.js
+    '''.format(sys.argv[0]))
+    )
+parser.add_argument('folders', metavar='folder', nargs='*',
+                    help='folder subset to run tests')
+parser.add_argument('-b', '--binary', metavar='binary', help='ch full path');
+parser.add_argument('-d', '--debug', action='store_true',
+                    help='use debug build');
+parser.add_argument('-t', '--test', action='store_true', help='use test build');
+parser.add_argument('--x86', action='store_true', help='use x86 build');
+parser.add_argument('--x64', action='store_true', help='use x64 build');
+args = parser.parse_args()
 
-test_all = True
-test_root = os.path.dirname(os.path.realpath(__file__))
-
-# ugly trick
-ch_path = os.path.join(os.path.dirname(test_root), "BuildLinux/ch")
 
-if not os.path.isfile(ch_path):
-    print "BuildLinux/ch not found. Did you run ./build.sh already?"
+test_root = os.path.dirname(os.path.realpath(__file__))
+repo_root = os.path.dirname(test_root)
+
+# arch: x86, x64
+arch = 'x86' if args.x86 else ('x64' if args.x64 else None)
+if arch == None:
+    arch = os.environ.get('_BuildArch', 'x86')
+
+# flavor: debug, test, release
+flavor = 'debug' if args.debug else ('test' if args.test else None)
+if flavor == None:
+    flavor = {'chk':'debug', 'test':'test', 'fre':'release'}\
+                    [os.environ.get('_BuildType', 'fre')]
+
+# binary: full ch path
+binary = args.binary
+if binary == None:
+    if sys.platform == 'win32':
+        binary = os.path.join(repo_root,
+            'Build/VcBuild/bin/{}_{}/ch.exe'.format(arch, flavor))
```

Comment:
This whole statement is less than 100 characters. I don't think we need to try so hard to fit within 80 columns. In python especially, I prefer not to split lines unless it specifically helps readability.